src/app/entity/entity.ts

```diff
@@ -23,25 +23,62 @@
  */
 export class Entity {
 
-  private _id: string;
+  /**
+   * The database id in the form <code>prefix:entityId</code>
+   */
+  private readonly _id: string;
 
+  /**
+   * The unique id of this entity.
+   */
+  private readonly entityId: string;
+
+  /**
+   * The prefix for this entity type.
+   */
+  private readonly prefix: string;
+
+  /**
+   * Creates an entity object with the given id. This id is final and won't be changeable after this object has been
+   * created.
+   *
+   * @param id a unique id for this entity
+   */
   constructor(id: string) {
-    this.setId(id);
+    this.entityId = id;
+    this.prefix = this.constructor.name;
+    this._id = this.prefix + ':' + this.entityId;
   }
 
+  /**
+   * Returns the id of this entity.
+   *
+   * Note that an id is final and can't be changed after the object has been instantiated, hence there is no
+   * <code>setId()</code> method.
+   *
+   * @returns {string} the unique id of this entity
+   */
+  public getEntityId(): string {
+    return this.entityId;
+  }
 
+  /**
+   * Returns the prefix which is used to categorize this entity in the database.
+   *
+   * <b>Important: Do not overwrite this method! Prefixes are handled internally now.</b>
+   *
+   * @returns {string} the prefix of this entity type (which is the class name).
+   */
   public getPrefix(): string {
-    return '';
+    return this.prefix;
   }
 
-  public getId(): string {
+  /**
+   * Returns the id of this entity including a prefix to store this entity in the database.
+   *
+   * @returns {string} the entity's id including a prefix in the form <code>prefix:entityId</code>
+   */
+  public getIdWithPrefix(): string {
```

Comment:
Hmm I'll have a look into that, not quite sure how to handle this in an elegant way.